typing/ctype.ml

```diff
@@ -208,7 +208,7 @@ let with_raised_nongen_level f =
 
 
 let reset_global_level () =
-  global_level := !current_level + 1
+  global_level := !current_level
```

Comment:
`global_level` is initialized at `1` above, should it change?

Reply:
This is actually the whole point of this change.
This makes no sense to create nodes at a level higher than `current_level`.
So the value of `global_level` was set based on the assumption that there would be immediately a cal to a function that raises `current_level` by 1, so that `global_level` would be equal to `current_level`.
This was fine in practice, but this still means that we have points in the code that break the invariant `!global_level <= !current_level`.
This PR avoids that by moving all calls to `reset_global_level` to places where `current_level` is already raised, and so we just set `global_level` to `current_level`.

Reply:
My question is as follows:

https://github.com/ocaml/ocaml/blob/8d8245d184d38a33cac88095a0d6cfc1ebc13d68/typing/ctype.ml#L148-L151

should we use `0` instead of `1` here?

Reply:
Sorry, I misunderstood your question.
Since the initial value of `global_level` is never used (all functions that translate type variables reset it first), this doesn't matter.
Eventually, we plan to remove those variables anyway.

Reply:
Maybe we could still consider moving the initial value to `0` to be consistent with the others, and with the proposed change?

Reply:
I requested this change as the only thing left before we can merge this PR.